RePak/include/assets/animation.h

```diff
@@ -0,0 +1,1028 @@
+	ACT_MP_JUMP_LAND_PRIMARY = 845,
+	ACT_MP_DEPLOYED_PRIMARY = 846,
+	ACT_MP_STAND_SECONDARY = 847,
+	ACT_MP_RUN_SECONDARY = 848,
+	ACT_MP_STAND_MELEE = 849,
+	ACT_MP_CROUCH_MELEE = 850,
+	ACT_MP_RUN_MELEE = 851,
+	ACT_MP_WALK_MELEE = 852,
+	ACT_MP_AIRWALK_MELEE = 853,
+	ACT_MP_CROUCHWALK_MELEE = 854,
+	ACT_MP_JUMP_MELEE = 855,
+	ACT_MP_JUMP_START_MELEE = 856,
+	ACT_MP_JUMP_FLOAT_MELEE = 857,
+	ACT_MP_JUMP_LAND_MELEE = 858,
+	ACT_MP_GESTURE_FLINCH_BACKWARDS = 859,
+	ACT_MP_GESTURE_FLINCH_LEFT = 860,
+	ACT_MP_GESTURE_FLINCH_FORWARDS = 861,
+	ACT_MP_GESTURE_FLINCH_RIGHT = 862,
+	ACT_MP_DEATH_CRUSH = 863,
+	ACT_MP_DEATH_GENERIC = 864,
+	ACT_MP_DEATH_CROUCH = 865,
+	ACT_MP_DEATH_WALK = 866,
+	ACT_MP_DEATH_WALK_FORWARD = 867,
+	ACT_MP_DEATH_WALK_BACKWARD = 868,
+	ACT_MP_DEATH_WALK_LEFT = 869,
+	ACT_MP_DEATH_WALK_RIGHT = 870,
+	ACT_MP_DEATH_RUN = 871,
+	ACT_MP_DEATH_RUN_FORWARD = 872,
+	ACT_MP_DEATH_RUN_BACKWARD = 873,
+	ACT_MP_DEATH_RUN_LEFT = 874,
+	ACT_MP_DEATH_RUN_RIGHT = 875,
+	ACT_MP_DEATH_SPRINT = 876,
+	ACT_MP_DEATH_SPRINT_FORWARD = 877,
+	ACT_MP_RUN_SPEEDPAINT_PRIMARY = 878,
+	ACT_MP_DROWNING_PRIMARY = 879,
+	ACT_MP_LONG_FALL_PRIMARY = 880,
+	ACT_MP_TRACTORBEAM_FLOAT_PRIMARY = 881,
+	ACT_MP_DEATH_CRUSH_PRIMARY = 882,
+};
+
+enum mstudioseqflags : uint32_t
+{
+	STUDIO_LOOPING = 0x0001,	// ending frame should be the same as the starting frame
+	STUDIO_SNAP = 0x0002,	// do not interpolate between previous animation and this one
+	STUDIO_DELTA = 0x0004,	// this sequence "adds" to the base sequences, not slerp blends
+	STUDIO_AUTOPLAY = 0x0008,	// temporary flag that forces the sequence to always play
+	STUDIO_POST = 0x0010,	//
+	STUDIO_ALLZEROS = 0x0020,	// this animation/sequence has no real animation data
+	STUDIO_FRAMEANIM = 0x0040,	// animation is encoded as by frame x bone instead of RLE bone x frame
+	STUDIO_CYCLEPOSE = 0x0080,	// cycle index is taken from a pose parameter index
+	STUDIO_REALTIME = 0x0100,	// cycle index is taken from a real-time clock, not the animations cycle index
+	STUDIO_LOCAL = 0x0200,	// sequence has a local context sequence
+	STUDIO_HIDDEN = 0x0400,	// don't show in default selection views
+	STUDIO_OVERRIDE = 0x0800,	// a forward declared sequence (empty)
+	STUDIO_ACTIVITY = 0x1000,	// Has been updated at runtime to activity index
+	STUDIO_EVENT = 0x2000,	// Has been updated at runtime to event index on server
+	STUDIO_WORLD = 0x4000,	// sequence blends in worldspace
+	STUDIO_NOFORCELOOP = 0x8000,	// do not force the animation loop
+	STUDIO_EVENT_CLIENT = 0x10000	// Has been updated at runtime to event index on client
+};
+
+#pragma pack(push, 2)
+
+// --- arig ---
+struct AnimRigHeader
+{
+	RPakPtr pSkeleton{};
+
+	RPakPtr pName{};
+
+	uint32_t Unk1;
+	uint32_t AseqRefCount;
+
+	RPakPtr pAseqRefs{};
+
+	uint64_t Padding;
+};
+
+// ANIMATIONS
+// --- aseq ---
+struct AnimHeader
+{
+	RPakPtr pAnimation{};
+
+	RPakPtr pName{};
+
+	uint64_t Unk1;
+	uint64_t Unk2;
+	uint64_t Unk3;
+
+	RPakPtr pModelHash{};
+};
```

Comment:
these structs are outdated. see below
`struct AnimRigHeader
{
	RPakPtr pRRIG;
    RPakPtr pName;

	int unk;

	int rseqCount;
	RPakPtr pRSEQ;

	int64 unk1;
};

struct ASeqHeader
{
	RPakPtr pData; // pointer to raw rseq.
	RPakPtr pName; // pointer to debug name, placed before raw rseq normally.

	// this can point to a group of guids and not one singular one.
	RPakPtr pModels;
	int modelCount;

    int reserved;

	// this can point to a group of guids and not one singular one.
	RPakPtr pSettings;
	int settingsCount;

    int reserved1;
};`